Fix dockerfile, use gr-gsm pre GNU Radio 3.8

BrandonFanti opened the pull request:
Since #74 has been open for 2 years now, and I kinda doubt anyone is working on updating the actual software to 3.8, at least this will function. 

Don't put it in master if there are plans to update?

Commit:
Fix dockerfile, use gr-gsm pre GNU Radio 3.8

Commit:
python3 and python3-distutils so simple_IMSI-catcher.py, can be run

also requirements.txt for pip (not necessary in docker)

Oros42 commented:
Why did you chose this commit https://gitea.osmocom.org/sdr/gr-gsm/commit/fa184a9447a90aefde2ca0dea1347b702551015d and not last ?

BrandonFanti commented:
I seem to have overlooked that you found your own [solution](https://github.com/Oros42/IMSI-catcher/issues/74#issuecomment-689122834) to running simple_IMSI-catcher.py with GNU-Radio 3.8 - so I guess this isn't necessary. 

I had a plethora of issues between last night and this morning, gsrm_livemon wouldn't run, complained about gnu-radio. Sorry for the lack of detail, I'm at work and on my windows partition, so I can't check atm.  

As @microwaven noticed in the issue I referenced - later versions of gr-gsm require 3.8, and this software was not cooperating, so I tied it down to that commit (That commit is the last one before GNU-radio 3.8 support was added) 

I can get your more detail later, or I may find that I was sleep deprived and this is dumb hacky fix to a problem that already has a much more elegant solution. TBD

BrandonFanti commented:
Right, here's what happened on my system running docker build. 

docker version

> Client: Docker Engine - Community
>  Version:           20.10.18
>  API version:       1.40
>  Go version:        go1.18.6
>  Git commit:        b40c2f6
>  Built:             Thu Sep  8 23:12:20 2022
>  OS/Arch:           linux/amd64
>  Context:           default
>  Experimental:      true

> Server: Docker Engine - Community
>  Engine:
>   Version:          19.03.13
>   API version:      1.40 (minimum version 1.12)
>   Go version:       go1.13.15
>   Git commit:       4484c46d9d
>   Built:            Wed Sep 16 17:01:25 2020
>   OS/Arch:          linux/amd64
>   Experimental:     false
>  containerd:
>   Version:          1.6.8
>   GitCommit:        9cd3357b7fd7218e4aec3eae239db1f68a5a6ec6
>  runc:
>   Version:          1.1.4
>   GitCommit:        v1.1.4-0-g5fd4c4d
>  docker-init:
>   Version:          0.18.0
>   GitCommit:        fec3683


Uname
> Linux {host} 4.19.0-22-amd64 #1 SMP Debian 4.19.260-1 (2022-09-29) x86_64 GNU/Linux


First run at docker build:
```
CMake Error at CMakeLists.txt:88 (find_package):
  Could not find a configuration file for package "Gnuradio" that is
  compatible with requested version "3.8".

  The following configuration files were considered but not accepted:

    /usr/lib/x86_64-linux-gnu/cmake/gnuradio/GnuradioConfig.cmake, version: 3.7.11.0
```

Fix in my case was to change to last commit before "GNU Radio 3.8 support" (was change I23dd638c8c48ed0a4b50559ce33fbd5e60b7dcbc)

So I rolled back to gr-gsm @ fa184a9447a90aefde2ca0dea1347b702551015d

Then in docker there was no python3, so I added that. I'm not sure for what I added distutils, but I didn't need it for anything I tested today. 

And I believe I referenced #74 because of @microwaven comment about the readme, but after reflection I don't think it was actually related to the root issue there, so not a fix.

Oros42 commented:
I didn't have time to work on this :-(
But I think the source of our problems is this :
`FROM ubuntu:bionic` in Dockerfile
https://github.com/Oros42/IMSI-catcher/blob/34d68e5cdc9b69e8d0feae70e06e44427e4370cd/Dockerfile#L1

`bionic` is so old.
Could you try with `FROM ubuntu:jammy` and replace `python` to `python3`?
